tests/test_extents.cpp

```diff
@@ -151,12 +151,59 @@ struct TestExtentsCompatCtors<std::tuple<
   static constexpr bool implicit_exts1_to_exts2 = ImplicitExts1ToExts2;
   static constexpr bool implicit_exts2_to_exts1 = ImplicitExts2ToExts1;
 
+  // Idx starts at rank()
+  template<class Exts, size_t Idx, bool construct_all, bool Static>
+  struct make_extents;
+
+  // Construct all extents - doesn't matter if idx is static or not
+  template<class Exts, size_t Idx, bool Static>
+  struct make_extents<Exts,Idx,true,Static>
+  {
+    template<class ... SizeTypes>
+    static Exts construct(SizeTypes...sizes) {
+      return make_extents<Exts, Idx-1,true,Static>::construct(Idx*5, sizes...);
+    }
+  };
+
+  // Construct dynamic extents - opnly add a parameter if the current index is not static
+  template<class Exts, size_t Idx>
+  struct make_extents<Exts,Idx,false,false>
+  {
+    template<class ... SizeTypes>
+    static Exts construct(SizeTypes...sizes) {
+      return make_extents<Exts, Idx-1,false,(Idx>1?Exts::static_extent(Idx-2)!=stdex::dynamic_extent:true)>::construct(Idx*5, sizes...);
+    }
+  };
+  template<class Exts, size_t Idx>
+  struct make_extents<Exts,Idx,false,true>
+  {
+    template<class ... SizeTypes>
+    static Exts construct(SizeTypes...sizes) {
+      return make_extents<Exts, Idx-1,false,(Idx>1?Exts::static_extent(Idx-2)!=stdex::dynamic_extent:true)>::construct(sizes...);
+    }
+  };
+
+  // End case
```

Comment:
Is it ever possible for `Idx` to be zero and `Static` to be `false`?  If so, then would you consider adding a base case for `Static=false` as well?

Reply:
no its not. The actual iteration with something happening is 1<=Idx<=Rank. I make sure that for Idx zero its always Static=true.